crates/steel-repl/src/highlight.rs

```diff
@@ -75,27 +93,66 @@ impl Highlighter for RustylineHelper {
 
         let mut line_to_highlight = line.to_owned();
 
-        let token_stream = TokenStream::new(line, true, None);
+        let mut token_stream = TokenStream::new(line, true, None).peekable();
 
         let mut ranges_to_replace: Vec<(std::ops::Range<usize>, String)> = Vec::new();
 
-        for token in token_stream {
+        let mut stack = vec![];
+        let mut cursor = None;
+
+        let mut paren_to_highlight = None;
+
+        while let Some(token) = token_stream.next() {
             // todo!()
 
             // if token.span().end() > pos {
             //     return self.highlighter.highlight(line, pos);
             // }
 
             match token.typ() {
-                // steel::parser::tokens::TokenType::OpenParen => todo!(),
-                // steel::parser::tokens::TokenType::CloseParen => todo!(),
+                TokenType::OpenParen(paren) if paren_to_highlight.is_none() => {
+                    if token.span.start == pos || (token.span.start == pos + 1 && cursor.is_none())
```

Comment:
We highlight the match of:
* The parenthesis right under the cursor if any, or
* The _closing_ parenthesis right to the _left_ of the cursor, if any, or
* The _opening_ parenthesis right to the _right_ of the cursor, if any.

The last point is a deviation from current behavior, where we always peek to the left, regardless of closing/opening. I find this slightly more... symmetrical? But I can easily revert to the current behavior.

Reply:
I think this is reasonable!